graphics.py

```diff
@@ -0,0 +1,61 @@
+import pandas as pd 
+import matplotlib.pyplot as plt 
+import os
+import re
+import sys
+import numpy as np
+
+def graphics(directorio):
+    resultados = []
+
+    for archivo in os.listdir(directorio):
+        if re.match(r'^threads_\d+_\d+_\d+_\d+\.csv$', archivo):
+            ruta_completa = os.path.join(directorio, archivo)
+            df = pd.read_csv(ruta_completa)
+
+            # Dividimos el archivo y extraemos Hilos y Método
+            partes = archivo.split('_')
+            num_hilos = int(partes[1])
+            metodo = partes[4].split('.')[0]
```

Comment:
Puedes aprovechar el mismo regex para hacer la separación en partes si utilizas grupos de captura
```suggestion
        match = re.match(r'^threads_(\d+)_(\d+)_(\d+)_(\d+)\.csv$', archivo)
        if match:
            ruta_completa = os.path.join(directorio, archivo)
            df = pd.read_csv(ruta_completa)

            # Dividimos el archivo y extraemos Hilos y Método
            num_hilos = match[1]
            # filas = match[2]
            # cols = match[3]
            metodo = match[4]
```